### PR TITLE
Add documentation for --build-profile warnings

### PR DESCRIPTION
**Pull request by cdmihai**
Added a page that details the incremental warnings and how users can address them.

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@blackdwarf can you review please?

File: Documentation/addressing-incremental-compilation-warnings.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Addressing Incremental Compilation Warnings
+===========================================
+
+Incremental compilation is unsafe when compilation relies on tools with potential side effects (tools that can cause data races or modify the state of other projects that have been deemed safe to skip compilation. Or tools that integrate timestamps or guids into the build output).
+
+The presence of such cases will turn off incremental compilation.
+
+Warning codes for project structures that are unsafe:
```
Comment (blackdwarf) [inline Documentation/addressing-incremental-compilation-warnings.md]: When you say "warning codes", this are the errors that the tool will write out?

**Comment by blackdwarf** [comment]
LGTM, with one minor nit.

File: Documentation/addressing-incremental-compilation-warnings.md (hunk 1 above)
Comment (cdmihai) [inline Documentation/addressing-incremental-compilation-warnings.md]: That's correct, they are the error codes the tool will print out. I'll update the doc to make it clear.

**Comment by cdmihai** [comment]
Addressed feedback

**Comment by cdmihai** [comment]
Addressed feedback:

- tests are run in parallel
- separated passing the bin dir to test in another PR: #748
- rewrote the incremental tests to use the new testing framework

Currently blocked by #748. Can merge once that is in.

**Pull request merged**